imports/client/components/SetupPage.tsx

```diff
@@ -453,6 +461,119 @@ const GoogleDriveTemplateForm = ({ initialDocTemplate, initialSpreadsheetTemplat
+    setEndpointUrl(e.currentTarget.value);
+  }, []);
+
+  const saveEndpointUrl = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    if (!endpointUrl) return;
+
+    setState({ submitState: SubmitState.SUBMITTING });
+    configureGoogleScriptUrl.call({ url: endpointUrl }, (error) => {
+      if (error) {
+        setState({ submitState: SubmitState.ERROR, error });
+      } else {
+        setState({ submitState: SubmitState.SUCCESS });
+      }
+    });
+  }, [endpointUrl]);
+
+  const shouldDisable = state.submitState === 'submitting';
+  return (
+    <>
+      {state.submitState === 'submitting' ? <Alert variant="info">Saving...</Alert> : null}
+      {state.submitState === 'success' ? <Alert variant="success" dismissible onClose={dismissAlert}>Saved changes.</Alert> : null}
+      {state.submitState === 'error' ? (
+        <Alert variant="danger" dismissible onClose={dismissAlert}>
+          Saving failed:
+          {' '}
+          {state.error.message}
+        </Alert>
+      ) : null}
+      <p>
+        Jolly Roger uses Google Apps Script to access APIs which are
+        otherwise not accessible through Google&apos;s public APIs, such as
+        inserting images into a Google Spreadsheet.
+      </p>
+      <p>
+        We need to offer users a way to insert images, because the insert
+        image dialog within Google Spreadsheets prevents itself from working
+        within an iframe.
+      </p>
+      <p>
+        This is a two-step process. First we need to create the project in
+        Google Apps Script (or update the source code, if it is out of date):
+      </p>
+      <p>
+        <Button variant="primary" onClick={ensureScript} disabled={shouldDisable}>Create (or update) Google Apps Script project</Button>
+      </p>
+      {app?.scriptId && (
+        <>
+          <p>
+            Next, you need to create a deployment. Unfortunately, Google does not
+            expose this via API, so you&apos;ll need to do it by hand:
+          </p>
+          <ul>
+            <li>Open the <a href={`https://script.google.com/home/projects/${app.scriptId}`} target="_blank" rel="noopener noreferrer">project</a></li>
+            <li>Click on the &quot;Deploy&quot; button in the top-right. Choose &quot;New deployment&quot;</li>
+            <li>Next to &quot;Select type&quot;, click the gear icon and choose &quot;Web app&quot;</li>
+            <li>Enter &quot;Jolly Roger Google Script Integration&quot; as the description. Make sure &quot;Execute as&quot; is set to &quot;me&quot; and &quot;Who has access&quot; is set to &quot;Anyone&quot; (this should be the default)</li>
+            <li>Click the &quot;Deploy&quot; button. You&apos;ll be prompted to authorize the app to access your Google Spreadsheets.</li>
+            <li>Once the deployment is finished, copy the URL under &quot;Web app&quot; and paste it below:</li>
```

Comment:
copy the URL, paste it below, and click the Save button

Reply:
👍🏻